lib/demo/accounts.ex

```diff
@@ -0,0 +1,349 @@
+defmodule Demo.Accounts do
```

Comment:
A question after reading through the code: why is this module called `Demo.Accounts`?

I'd expect it to be simply called `Demo.Users`, as almost all functions and/or first function arguments have "user" in them somewhere. Also, I do not really see any usage of the term "account" in function arguments or otherwise. Lastly, in the `DemoWeb` context, most routes have "user" in their paths, and "user" generally seems to be the dominant term used everywhere.

I could every well be missing something here! Just wanted to ask as there may be an opportunity to simplify function naming (e.g. removing some "user_" prefixes in this module) and avoiding possible terminology confusion between "Account" and "User".

Off-topic: Great work on this! Feels like a great approach in general. (Speaking here as a former Devise user who has been trying out multiple Devise-like frameworks with Elixir/Phoenix, leading to similar experiences as outlined in the Dashbit blogpost 😄)

Reply:
The module name is whatever you specify during generation for the context. So `mix phx.gen.auth Users User users ...extrafields...` would do as you say.

Reply:
The [Phoenix docs for Contexts](https://hexdocs.pm/phoenix/contexts.html) may be (one of) the reasons why this was chosen. If nothing else it could allow people new to the framework to connect the documentation examples.